### PR TITLE
Add Cancellation player to roulette game.

### PR DESCRIPTION
**Pull request by FarzanRashid**
This PR adds Cancellation player to the roulette game. This PR also includes tests for the PlayerCancellation class.

### CODE REVIEW

File: src/players/cancellation_player.py [hunk 1]
```diff
@@ -0,0 +1,86 @@
...
+        """
+        :param bet: The bet which won
+
+        Uses the superclass method to update the stake with an amount won. It then removes the first
+        and last element from **sequence**.
+        """
+
+        super().win(bet)
+        self.sequence.pop()
+        self.sequence.pop(0)
```
Comment (FarsanRashid) [inline src/players/cancellation_player.py:66]: need to check  sequence emptiness.

File: src/players/cancellation_player.py
Comment (FarsanRashid) [inline src/players/cancellation_player.py:1]: change file name to cancellation.py

File: tests/test_cancellation_player.py [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+
+
+class TestPlayerCancellation(TestCase):
+    def setUp(self):
+        self.table = Table()
+        self.player_cancellation = PlayerCancellation(self.table)
+
+    def test_resetSequence_resets_sequence(self):
+        self.player_cancellation.sequence = []
+        expected_seq_after_resetSequence = [1, 2, 3, 4, 5, 6]
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:15]: renaming to `expected_seq` is enough.

File: tests/test_cancellation_player.py [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+        self.player_cancellation.resetSequence()
+        self.assertEqual(
+            self.player_cancellation.sequence, expected_seq_after_resetSequence
+        )
+
+    def test_lose_adds_bet_amount_to_sequence(self):
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:21]: better rename to `bet_amount_added_to_sequence_after_lose`. This way test name represents a behavior. The current test name shows which method (lose) updates the sequence, this is implementation details.

File: tests/test_cancellation_player.py [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+        bet = Bet(10, Outcome("Red", 1))
+        self.assertNotIn(bet.amount, self.player_cancellation.sequence)
+
+        self.player_cancellation.lose(bet)
+        self.assertIn(bet.amount, self.player_cancellation.sequence)
+
+    def test_win_removes_elements_from_sequence(self):
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:28]: same as renaming of `test_lose...`

File: tests/test_cancellation_player.py [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        expected_seq_before_win = [1, 2, 3, 4, 5, 6]
+        self.assertEqual(expected_seq_before_win, self.player_cancellation.sequence)
+
+        expected_seq_after_win = [2, 3, 4, 5]
+
+        self.player_cancellation.win(bet)
+
+        self.assertEqual(expected_seq_after_win, self.player_cancellation.sequence)
+
+    def test_placeBets_reduces_stake(self):
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:39]: better rename to `test_placing_bet_reduces_stake` which resonates a behavior.

File: tests/test_cancellation_player.py [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+        self.player_cancellation.stake = 100
+        expected_stake_after_bet = 93
+
+        self.player_cancellation.placeBets()
+        self.assertIs(self.player_cancellation.stake, expected_stake_after_bet)
+
+    def test_placeBets_places_bet(self):
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:46]: better rename to `test_bets_are_placed`.

File: tests/test_cancellation_player.py [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+    def test_player_plays_when_stake_more_than_bet_amount(self):
+        self.player_cancellation.bet_amount = 9
+
+        self.player_cancellation.stake = 10
+        self.assertTrue(self.player_cancellation.playing())
+
+        self.player_cancellation.stake = 8
+        self.assertFalse(self.player_cancellation.playing())
+
+    def test_player_plays_when_roundstoGo_not_zero(self):
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:65]: better rename to `test_player_plays_when_rounds_remaining`. which variable(roundstoGo) controls this behavior is implementation details and should not be part of test.

File: tests/test_cancellation_player.py [hunk 8]
```diff
@@ -0,0 +1,88 @@
...
+
+        self.player_cancellation.sequence = [1]
+        self.assertFalse(self.player_cancellation.playing())
+
+    def test_playing_resets_sequence_if_player_not_playing(self):
+        self.player_cancellation.sequence = []
+        self.player_cancellation.stake = 0
+        expected_sequence_after_playing = [1, 2, 3, 4, 5, 6]
+
+        self.player_cancellation.playing()
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:85]: asserting `playing` equals false makes the test more expressive.

File: tests/test_cancellation_player.py [hunk 9]
```diff
@@ -0,0 +1,88 @@
...
+        self.assertFalse(self.player_cancellation.playing())
+
+    def test_player_plays_when_have_enough_sequence(self):
+        self.player_cancellation.sequence = [1, 2, 3]
+        self.assertTrue(self.player_cancellation.playing())
+
+        self.player_cancellation.sequence = [1]
+        self.assertFalse(self.player_cancellation.playing())
+
+    def test_playing_resets_sequence_if_player_not_playing(self):
```
Comment (FarsanRashid) [inline tests/test_cancellation_player.py:80]: better rename to `test_sequence_is_reset_when_player_is_not_playing`.

File: src/players/cancellation_player.py
Comment (FarzanRashid) [reply src/players/cancellation_player.py:1]: Done.

File: tests/test_cancellation_player.py [hunk 10]
```diff
@@ -0,0 +1,88 @@
...
+from outcome import Outcome
+from players.cancellation_player import PlayerCancellation
+
+
+class TestPlayerCancellation(TestCase):
+    def setUp(self):
+        self.table = Table()
+        self.player_cancellation = PlayerCancellation(self.table)
+
+    def test_resetSequence_resets_sequence(self):
```
[the remark this answers is not in the document]
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:13]: Done.

File: tests/test_cancellation_player.py (hunk 2 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:15]: Done.

File: tests/test_cancellation_player.py (hunk 3 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:21]: Done.

File: tests/test_cancellation_player.py (hunk 4 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:28]: Done.

File: tests/test_cancellation_player.py (hunk 5 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:39]: Done.

File: tests/test_cancellation_player.py (hunk 6 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:46]: Done.

File: tests/test_cancellation_player.py (hunk 7 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:65]: Done.

File: tests/test_cancellation_player.py (hunk 8 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:85]: Done.

File: tests/test_cancellation_player.py (hunk 9 above)
Comment (FarzanRashid) [reply tests/test_cancellation_player.py:80]: Done.

**Pull request merged**